### PR TITLE
Added several unit tests and refactoring of code; ENT-942

### PR DESCRIPTION
**Pull request by jirihnidek**
* Added several unit tests
* Refactored code to make it more testable
* It is not possible to write unit tests for some functions, because unit test framework provided by glib does not support mocks

### CODE REVIEW

**Comment by jirihnidek** [comment]
I will create another PR for issue described by @jrohel. The PR will be created on top of this branch.

**Comment by jirihnidek** [comment]
@candlepin-pull-request-bot please retest this

**Comment by jirihnidek** [comment]
@candlepin-pull-request-bot Please retest this.

File: src/dnf-plugins/product-id/product-id.c [hunk 1]
```diff
@@ -182,7 +192,12 @@ int pluginHook(PluginHandle *handle, PluginHookId id, void *hookData, PluginHook
 
     if (id == PLUGIN_HOOK_ID_CONTEXT_TRANSACTION) {
         // Get DNF context
-        DnfContext *dnfContext = handle->initData;
+        DnfContext *dnfContext = handle->context;
+        if (dnfContext == NULL) {
+            error("Unable to get get dnf context");
```
Comment (cnsnyder) [inline src/dnf-plugins/product-id/product-id.c]: One too many "get".

**Pull request merged**